# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/Realm.cpp
# Commit: 33dcf5c2e643
# Full Hash: 33dcf5c2e643b94e90a0a197c99112a080501a12
# Author: Alexandra Borovova <aborovova@mozilla.com>
# Date: 2025-08-20 09:57:21
# Regressor Bug: 1980211
# File Overlap Count: 2
# Description:
#   Bug 1980211 - Attach locale override to Realm instead of setting it in the process. r=dom-core,smaug,tschuster
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D260169
# ==============================================================================

diff -r 2cb7ca176216 -r 33dcf5c2e643 js/src/vm/Realm.cpp
--- a/js/src/vm/Realm.cpp	Tue Aug 19 16:57:59 2025 +0000
+++ b/js/src/vm/Realm.cpp	Tue Aug 19 17:01:50 2025 +0000
@@ -530,6 +530,10 @@
 void Realm::clearScriptLCov() { zone()->clearScriptLCov(this); }
 
 const char* Realm::getLocale() const {
+  if (RefPtr<LocaleString> locale = behaviors_.localeOverride()) {
+    return locale->chars();
+  }
+
   if (RefPtr<LocaleString> locale = creationOptions_.locale()) {
     return locale->chars();
   }